src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -1200,7 +1200,7 @@ static void do_nothing(grpc_exec_ctx *exec_ctx, void *arg, grpc_error *error) {}
 
 static void log_metadata(const grpc_metadata_batch *md_batch, uint32_t id,
                          bool is_client, bool is_initial) {
-  for (grpc_linked_mdelem *md = md_batch->list.head; md != md_batch->list.tail;
+  for (grpc_linked_mdelem *md = md_batch->list.head; md != NULL;
```

Comment:
Yes, this was a bug. It was indeed skipping the last element. Nobody noticed because this was just logging and only used in debugging. @ctiller :  I should have mentioned this to you but it didn't seem too important.